akka-actor/src/main/scala/akka/io/dns/CachePolicy.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns
+
+import akka.annotation.ApiMayChange
+
+import scala.concurrent.duration.{ Duration, FiniteDuration, _ }
+
+object CachePolicy {
+
+  @ApiMayChange
+  sealed trait CachePolicy
+  @ApiMayChange
+  case object Never extends CachePolicy
+  @ApiMayChange
+  case object Forever extends CachePolicy
+  @ApiMayChange
+  final class Ttl(val value: FiniteDuration) extends CachePolicy {
```

Comment:
The constructor should be (Scala) `private`:
```suggestion
  final class Ttl private (val value: FiniteDuration) extends CachePolicy {
```